Add PR template specific to the schema soft freeze (`berkeley-schema-fy24`) (#217)
microbiomedata/berkeley-schema-fy24: .github/PULL_REQUEST_TEMPLATE/pull_request_template_berk.md

# Guidelines

## _Soft_ Schema Freeze

The `nmdc-schema` and `berkeley-schema-fy24` schemas are under a soft freeze, which means changes **should not** be made that have any downstream implications. To ensure this, all PRs created creating during the freeze will be closely reviewed with **every** component of the NMDC system in mind.

## Reviewers

To ensure no changes are made unexpectedly, PR creators will request reviews from _all_ [Berkeley Schema Roll Out task coordinators](https://docs.google.com/document/d/1XXN1YuaBuSkxPXeiLKm5YxYzXTamBPQrzzeLhlh7PWs/edit#heading=h.u52g8v319adh).

We expect task coordinators to review PRs and provide feedback/approval within 1 week of when they are identified as reviewers. 

PRs will **NOT** be merged until all task coordinators (or one of their delegates) have approved. 

Expedition, questions, and discussion can happen at any meeting.

Delays in review & merging should be addressed in meetings or with NMDC leadership.

# PR Information

## What type of PR is this? (check all applicable)

- [ ] Refactor
- [ ] Feature
- [ ] Bug Fix
- [ ] Optimization
- [ ] Documentation

> - [ ] Schema change: Content
>   - `slot` or `class` name
>   - changes to a `slot`'s `multivalued` state. `true` means the schema expects a list of values and `false` means  a single value is expected.
>   - any changes to the ways `slot`s are assigned to any `class`es
>   - creating a new `slot` or `class`
>   - additions, removals or changes to the `permissible_values` in an `enum`
> - [ ] Schema change: Cleanup
>   - descriptions of `slot`s, `class`es, or `enum`s
>   - removal of unused, commented, or invalid code
>   - updated `mappings` of `slot`s, `class`es, or `slot`s to ontologies
>   - addition of a new `enum` in the absence of making that `enum` the `range` for any existing `slot`

Reviewer: I made two groups:

- Structure and content
- Cleanup and preparation

I also standardized the phrasing of the sub-bullets.

```suggestion
- [ ] Schema change: Structure and content
  - created, updated, or deleted a `class`, `slot`, or `enum`
  - changed whether a `slot` is `multivalued`
  - changed the way a `slot` is assigned to a `class`
  - changed the `permissible_values` of an `enum`
  - _etc._
- [ ] Schema change: Cleanup and preparation
  - updated the description of a `class`, `slot`, or `enum`
  - updated the `mappings` of a `class`, `slot`, or `enum` to an ontology
  - added an `enum` for _future_ use (it is not in the `range` of any `slot`)
  - _etc._
```

Reviewer: Some of the attendees of today's metadata meeting discussed this suggestion today and nobody expressed objections to it.

Also, some of them clarified that they see the first group as "more likely to have downstream implications" and the second group "less likely to have downstream implications" (relative to the other group).

## Related Issues

> All PRs should relate to or fix an issue(s). Please identify the issue(s) below.

- Related Issue(s): #
- Fixes: #

## Did you add/update any tests?

- [ ] Yes
- [ ] No _(Add a justification below)_
- [ ] I need help with writing tests

## Could this schema change make it so any valid data becomes invalid?

> This is a question about what the schema allows. It is not a question about what happens to exists in the NMDC database right now.
> 
> Example: If, in this PR branch, you renamed a `slot` from `foo` to `foo_bar`, the answer to this question would be "yes," **even if** nothing in the NMDC database _currently_ uses the `foo` `slot`.
>
> More examples: `slot` or `class` name changes, changes to a `slot`'s `multivalued` state, changes to a `slot`'s `range` (e.g. string to integer), changes to `slot` assignments to `class`es, changes to an `enum`'s `permissible_values`

- [ ] Yes _(A migrator is required)_
- [ ] No
- [ ] I need help determining this

## If you answered "Yes", does this PR branch include that migrator?

- [ ] Yes
- [ ] No, this PR is incomplete and I need help writing the migrator

## Does this PR have any downstream implications?

> > **Ideally**, no schema changes will be performed that have downstream implications.

Reviewer: I propose removing this line as I think it's redundant with the intro. Also, phrasing of a "schema change" being "performed" (as opposed to "made" or "applied") doesn't sound natural to my ears.

> > **Ideally**, no schema changes will be performed that have downstream implications.
> >

Reviewer: ```suggestion

```

I propose removing this as (1) I think it's redundant with the intro, and (2) saying a "schema change" will be "performed" (as opposed to "made" or "applied") doesn't sound natural to my ears.

> > Examples of downstream changes: any change that requires a change to workflows, workflow automation, the Mongo-to-Postgres ingest process, Jupyter notebooks, the Runtime, etc.

Reviewer: ```suggestion
> Examples: any change that requires a change to workflows, workflow automation, the Mongo-to-Postgres ingest process, Jupyter notebooks, the Runtime, etc.
```

> - [ ] Yes _(Add a justification below)_

Reviewer: ```suggestion
- [ ] Yes _(Explain below)_
```

We expect task coordinators to review PRs and provide feedback/approval within 1 week of when they are identified as reviewers. 

PRs will **NOT** be merged until all task coordinators (or one of their delegates) have approved. 

Expedition, questions, and discussion can happen at any meeting.

Delays in review & mergeing should be address in meetings or with NMDC Leadership.

# PR Information

## What type of PR is this? (check all applicable)

- [ ] Refactor
- [ ] Feature
- [ ] Bug Fix
- [ ] Optimization
- [ ] Documentation
- [ ] Schema change: Content
  - slot or class name
  - slot multiplicity changes (from a single value to a list or vice versa)
  - any changes to the ways slots are assigned to any classes
  - creating a new slot/Class
  - Enum value changes
- [ ] Schema change: Cleanup
  - descriptions of slots, classes, or enums
  - removal of unused, commented, or invalid code
  - updated mappings of terms, classes, or slots to ontologies

>   - added an additional Enum to accommodate new or future metadata

Reviewer: FYI: I'm working on a proposal for these two bullet points (and their contents). I'll post it here once I'm done.

Reviewer: ```suggestion
- [ ] Schema change — Structure and content; for example:
  - adds or renames a class, slot, or Enum
  - changes the multiplicity of a slot (e.g. string ↔ list)
  - changes how a slot is used by a class
  - adds or updates an Enum value
- [ ] Schema change — Cleanup and documentation; for example:
  - updates the description of a class, slot, or Enum
  - updates comments
  - deletes unused code
```

Reviewer: I'll close this proposal since the code is was based on has been changed via a different means. I will re-express the proposal in a few minutes.

- [ ] Changes to any files or directories in `src/schema`
     
## Description

> _PRs should be [small and concise](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/getting-started/best-practices-for-pull-requests#write-small-prs)._
>
> Aim to create small, focused pull requests that fulfill a single purpose. Smaller pull requests are easier and faster to review and merge, leave less room to introduce bugs, and provide a clearer history of changes.

- Replace this text with a description of what this PR branch contains. Please keep in mind that all reviewers will be reading this description. Example: "In this branch, I..."

## Related Issues

> > All PRs should relate to or fix an issue(s). Please identify the issue(s) below.

Reviewer 2: Ideally, the issue will appear in the right hand side of the PR screen. I think the section is called "development"

That link is created automatically when issues are linked to branches
